app/src/main/java/ru/yandex/architectureproject/presentation/viewmodel/TaskViewModel.kt

```diff
@@ -30,12 +22,39 @@ class TaskViewModel(
     private val _state = MutableStateFlow<TaskState>(TaskState.Loading)
     val state: StateFlow<TaskState> = _state.asStateFlow()
 
+    private val taskCompletionsJobs = mutableMapOf<Int, Job>()
+
     init {
         reduce(TaskAction.LoadTasks)
     }
 
     fun reduce(action: TaskAction) {
-        // TODO: Здесь должна быть обработка действий
+        val job = viewModelScope.launch {
+            when (action) {
+                is TaskAction.LoadTasks -> loadTasks()
+                is TaskAction.AddTask -> addTaskUseCase(action.task)
+                is TaskAction.DeleteTask -> deleteTaskUseCase(action.taskId)
+                is TaskAction.UpdateTaskStatus -> {
+                    if (action.isDone) {
+                        completeTaskUseCase(action.taskId)
+                    } else {
+                        incompleteTaskUseCase(action.taskId)
+                    }
+                }
+            }
+        }
+
+        if (action is TaskAction.UpdateTaskStatus) {
+            if (action.isDone) {
+                taskCompletionsJobs[action.taskId] = job
+            } else {
+                taskCompletionsJobs[action.taskId]?.cancel()
```

Comment:
Отменяемую джобу так же стоит удалять из мапы:
```suggestion
                taskCompletionsJobs[action.taskId]?.remove()?.cancel()
```